app/controllers/application_controller.rb

```diff
@@ -959,7 +959,7 @@ def listicon_item(view, id = nil)
   # Return the icon classname for the list view icon of a db,id pair
   # this always supersedes listicon_image if not nil
   def listicon_icon(item)
-    item.decorate.try(:fonticon) if item.decorator_class?
+    item.decorate.try(:fonticon)
```

Comment:
With my other comments addressed, this now becomes a non-patched call, entirely dependent on the UI's own code, with `MiqDecorator.decorate(item)`. You of course can feel free to create a top level controller helper like `decorate_model` that does the same thing without `MiqDecorator` explicitly.